### ISSUE TITLE
Provide minDepth option.

### ISSUE THREAD

**Issue by meblum** [abhinav/goldmark-toc#40]
I have my headers as follows:
```
# h1
## h2
### h3 one
### h3 two
### h3 three
```
I don't want my main page title to be part of the toc. Is there a way to create a toc for h3 headers only?

Desired output:
```
<ul>
<li><a href="#h3-one">h3 one</a></li>
<li><a href="#h3-two">h3 two</a></li>
<li><a href="#h3-three">h3 three</a></li>
</ul>
```

**Comment by abhinav** [comment]
A MinDepth option makes sense. I'm in favor of it.
@meblum if you'd like to take a stab at it, I'll be happy to provide guidance. It'll look much the same as the MaxDepth option.
Otherwise, I'll try to get to it in the near future, but I can't promise how soon.

**Comment by meblum** [comment]
Thanks, I appreciate your work. Will have a look tomorrow.

Since we are at it, I just realized that it will render empty list items when a "parent" header is not present. For example, `### h3` renders
```
<h1>Table of Contents</h1>
<ul>
<li>
<ul>
<li>
<ul>
<li>
<a href="#h3">h3</a></li>
</ul>
</li>
</ul>
</li>
</ul>
<h3 id="h3">h3</h3>
```
Would probably make more sense to render existing items only, as the browser will happily render these empty items like so:
![image](https://github.com/abhinav/goldmark-toc/assets/33358938/1ce432a4-d285-47cd-9832-9561027877c3)

**Comment by abhinav** [comment]
Good catch; I ran into that myself recently.
I thought it might make sense to also add a "Compact" option (or make that the default).
I'll pull this part into a separate issue.

**Comment by abhinav** [comment]
Released in v0.7!
